### ISSUE TITLE
Samples shared between "Normal" and "Tumor" are processed twice by batch command

### ISSUE THREAD

**Issue by tskir** [etal/cnvkit#48]
Hi again,

If a BAM file is present in both "Normal" and "Tumor" collections supplied to `cnvkit batch` command, bin coverages for it (`sample.[anti]targetcoverage.cnn`) will be calculated twice, slowing down the entire process twofold.

It is true that "Normal" and "Tumor" sets normally do not overlap at all. It is often useful, however, to run all present (presumably normal) samples against themselves, for example to construct a reference for later use and to identify noisy / otherwise unsuitable samples in the process.

Yours
Kirill

**Comment by etal** [comment]
Good point. It would be nice for the batch command to behave more like `make`. In the meantime I suggest running the analysis like this:

1. `batch` with a flat reference on all "normal" samples, treating them as tumor samples:

    `cnvkit.py batch *_Normal.bam -n -d normals/ ...`
    `cnvkit.py heatmap normals/*.cns -o heatmap_normal_flatreference.pdf`

2. Examine the heatmap of the output .cns files to see which "normal" samples have problematic CNVs and should be excluded.
3. Build a reference from the selected normals:

    `cnvkit.py reference normals/{1,2,4,5,7}_Normal.*.cnn -f hg19.fa -o real_reference.cnn`

4. Use that reference to process the tumor samples

    `cnvkit.py batch *_Tumor.bam -r real_reference.cnn -d normals/ ...`

Alternatively, you can process all of the BAMs as normals, or tumors with a flat reference, and use the `reference` and `fix` commands directly to finish the analysis.

**Comment by tskir** [comment]
Thanks for the detailed instructions, Eric. I'll be sure to make use of them.

**Comment by etal** [comment]
See also #81 - now an error is raised instead of processing the same sample twice. (Still suboptimal, but we're working on it.)
